pymc_marketing/mmm/marginal_effects.py

```diff
@@ -0,0 +1,173 @@
+#   Copyright 2022 - 2025 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Counterfactual sweeps for Marketing Mix Models (MMM)."""
+
+import arviz as az
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+
+class CounterfactualSweep:
+    """CounterfactualSweep class is used to perform counterfactual analysis on MMM's."""
+
+    def __init__(
+        self,
+        mmm,
+        X: pd.DataFrame,
+        predictors: list[str],
+        sweep_values: np.ndarray,
```

Comment:
Think the npt.NDArray typing would be best to use

https://numpy.org/doc/stable/reference/typing.html